clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -770,13 +770,13 @@ public FetchManagerMetrics(Metrics metrics, String metricGrpPrefix) {
         }
 
         public void recordTopicFetchMetrics(String topic, int bytes, int records) {
-            Map<String, String> metricTags = new HashMap<>();
-            metricTags.put("topic", topic.replace(".", "_"));
-
             // record bytes fetched
             String name = "topic." + topic + ".bytes-fetched";
             Sensor bytesFetched = this.metrics.getSensor(name);
             if (bytesFetched == null) {
+                HashMap<String, String> metricTags = new HashMap<>();
+                metricTags.put("topic", topic.replace('.', '_'));
```

Comment:
A couple of things (applies to the same code a few lines below):

* Is there a reason you changed `Map<String, String>` to `HashMap<String, String>`?
* We can pass `1` to the `HashMap` constructor to avoid allocating a 16 entries array when we `put` `topic`.